dot/parachain/bitfield-distribution/bitfield_distribution_test.go

```diff
@@ -227,13 +233,16 @@ func TestBitfieldDistribution_FilterByPeerVersion(t *testing.T) {
 				p5: uint32(1),
 			},
 			targetVersion: uint32(4),
-			result:        []peer.ID{},
+			result:        map[peer.ID]struct{}{},
 		},
 	}
 
 	for _, testcase := range testcases {
-		re := filterByPeerVersion(testcase.peers, testcase.targetVersion)
-		assert.Equal(t, testcase.result, re, testcase.name)
+		m := make(map[peer.ID]struct{})
+		for _, val := range filterByPeerVersion(testcase.peers, testcase.targetVersion) {
+			m[val] = struct{}{}
+		}
+		assert.Equal(t, true, reflect.DeepEqual(testcase.result, m), testcase.name)
```

Comment:
Is the `DeepEqual()` really necessary? I thought `assert.Equal()` essential does the same.

Reply:
are they? I was just trying to make sure they are the same for sure, like double check